test/lib/jdk/test/lib/util/ForceGC.java

```diff
@@ -23,68 +23,57 @@
 
 package jdk.test.lib.util;
 
-import java.lang.ref.Cleaner;
-import java.util.concurrent.CountDownLatch;
-import java.util.concurrent.TimeUnit;
+import java.lang.ref.PhantomReference;
+import java.lang.ref.Reference;
+import java.lang.ref.ReferenceQueue;
 import java.util.function.BooleanSupplier;
 
 /**
  * Utility class to invoke System.gc()
  */
 public class ForceGC {
-    private final static Cleaner cleaner = Cleaner.create();
-
-    private final CountDownLatch cleanerInvoked;
-    private Object o;
-    private int gcCount = 0;
-
-    public ForceGC() {
-        this.o = new Object();
-        this.cleanerInvoked = new CountDownLatch(1);
-        cleaner.register(o, cleanerInvoked::countDown);
-    }
-
-    private void doit(int iter) {
-        try {
-            for (int i = 0; i < 10; i++) {
-                System.gc();
-                gcCount++;
-                if (cleanerInvoked.await(100L, TimeUnit.MILLISECONDS)) {
-                    return;
-                }
-            }
-        } catch (InterruptedException unexpected) {
-            throw new AssertionError("unexpected InterruptedException");
-        }
-    }
+    // The jtreg testing timeout factor.
+    private static final double TIMEOUT_FACTOR = Double.valueOf(
+            System.getProperty("test.timeout.factor", "1.0"));
```

Comment:
If you don't mind the dependency, you can use jdk.test.lib.Utils.TIMEOUT_FACTOR
or `Utils.adjustTimeout(xx)` in the usage below.